backend/workflow_manager/endpoint/destination.py

```diff
@@ -180,24 +179,30 @@ def handle_output(
             else:
                 self.insert_into_db(input_file_path=input_file_path)
         elif connection_type == WorkflowEndpoint.ConnectionType.API:
+            file_history = None
+            if use_file_history:
+                file_history = FileHistoryHelper.get_file_history(
+                    workflow=workflow, cache_key=file_hash.file_hash
+                )
```

Comment:
@chandrasekharan-zipstack also if we keep the file_history fetching only here won't we skip this step in other cases?

Should we simply make the file_history None here if not use_file_hisotry?